src/util/misc/xml.ts

```diff
@@ -0,0 +1,121 @@
+import {  } from "xmldom";
+import { child } from "event-stream";
+
+export function appendToTestNameTransformation(xml: Document, text: string) {
+  let testCases: Node[] = collectAllElements(xml, "test-case");
+  testCases.forEach((testCase: Node) => {
+    let name: Attr = testCase.attributes.getNamedItem("name");
+    if(name) {
+      name.value = `${name.value}${text}`; // TODO: value or nodeValue?
+    }
+  });
+}
+
+export function removeIgnoredTransformation(xml: Document) {
+  var testResults: Node[] = collectAllElements(xml, "test-results");
+  testResults.forEach((testResult: Node) => {
+    let ignoredAttr: Attr = testResult.attributes.getNamedItem("ignored");
+    if (ignoredAttr) {
+      let notRunAttr: Attr = testResult.attributes.getNamedItem("not-run");
+      if (notRunAttr) {
+        let notRun = Number(notRunAttr.value);
+        let ignored = Number(ignoredAttr.value);
+        notRunAttr.value = String(notRun - ignored);
+      }
+      ignoredAttr.value = "0";
+    }
+  });
+
+  var nodes: Node[] = collectAllElements(xml, "test-case");
+  nodes.forEach((node: Node) => {
+    let resultAttr = node.attributes.getNamedItem("result");
+    if(resultAttr && resultAttr.value === "Ignored") {
+      node.parentNode.removeChild(node);
+    }
+  });
+}
+
+export function removeEmptySuitesTransformation(xml: Document) {
+  var nodes: Node[] = collectAllElements(xml, "test-suite");
+  nodes.forEach((node: Node) => {
+    if (countChildren(node) <= 1) {
+      node.parentNode.removeChild(node);
+    }
+  });
+}
+
+export function combine(xml1: Document, xml2: Document): Document {
+  combineTestResultsAttribute(xml1, xml2, "total");
```

Comment:
@erychagov - would it be possible to add some tests for this new util file? I'd be particularly interested in seeing how it behaves when the report xml files are malformed (e.g. missing expected elements etc).

Depends a little how long that would take so please push back if you don't think this would be valuable.

Reply:
This is a ton of test cloud specific stuff, not something that would be used across the entire CLI. It doesn't belong under util/misc, it should be in with your test cloud commands.

Reply:
@owenniblock 
I've added tests

Reply:
@erychagov - I think this was deleted previously. but added back in by a bad merge. Or maybe it was never deleted - but the former seems more likely. We should delete this one?